### PR TITLE
RB 55 Uploading the book cover to Cloudinary

### PR DESCRIPTION
**Pull request by vrudnitskaya**
## Description
The user has two options for uploading. The first one is by providing the URL of the book cover. The second one is by providing a file.
## Screenshots
The user may also choose not to provide either a file or a URL, in which case the book will have a default image.
book-created-default-image

If the user edits a book and provides a new file or URL, the old image is deleted from the cloud.
update-book-with-file

terminal-deleting-image

When a book is deleted from the database, its cover image is also removed from Cloudinary.
delete-book-delete-image

### CODE REVIEW

File: src/controllers/bookController.js [hunk 1]
```diff
@@ -122,8 +124,14 @@ const createBook = async (req, res, next) => {
 
     bookData.createdBy = req.user.userId;
     bookData.language = req.body.language || 'English';
-    bookData.coverImageUrl =
-      coverImageUrl || process.env.DEFAULT_COVER_IMAGE_URL;
+
+    try {
+      const imageData = await uploadImage(req.file, coverImageUrl);
+      bookData.coverImageUrl = imageData.coverImageUrl;
```
Comment (benochi) [inline src/controllers/bookController.js:130]: Just food for thought, we would want https:// Urls for security.

File: src/config/cloudinary.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const cloudinary = require('cloudinary').v2;
+
+cloudinary.config({
+  cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
+  api_key: process.env.CLOUDINARY_API_KEY,
+  api_secret: process.env.CLOUDINARY_API_SECRET,
+});
+
+const DEFAULT_IMAGE_URL =
```
Comment (benochi) [inline src/config/cloudinary.js]: These should be in an .env but you can fix that later. 

Just imagine your application becomes huge and you need to scale it, you would want image sources at the CDN location that is serving your web application.

File: src/middleware/uploadFile.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const multer = require('multer');
+const { BadRequestError } = require('../errors');
+
+const MAX_FILE_SIZE = 1 * 1024 * 1024;
+const ALLOWED_FILE_TYPES = ['image/jpeg', 'image/png', 'image/webp', 'image/bmp'];
+
+const storage = multer.memoryStorage();
+
```
Comment (benochi) [inline src/middleware/uploadFile.js:8]: Food for thought: we don't reject corrupted files or any specific encodings, which is totally fine for this application, but I want you know have that in your brains.

**Review by benochi — APPROVED** [review]
Looks good :)

**Review by Tanyaa-a — APPROVED** [review]

**Pull request merged**